eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReader.java

```diff
@@ -0,0 +1,147 @@
+
+    static final String STUDY_ID_KEY = "SID";
+
+    private String assemblyAccession;
+
+    private MongoClient mongoClient;
+
+    private String database;
+
+    private MongoCursor<Document> cursor;
+
+    public AccessionedVariantMongoReader(String assemblyAccession, MongoClient mongoClient,
+                                         String database) {
+        this.assemblyAccession = assemblyAccession;
+        this.mongoClient = mongoClient;
+        this.database = database;
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        MongoDatabase db = mongoClient.getDatabase(database);
+        MongoCollection<Document> collection = db.getCollection(DBSNP_CLUSTERED_VARIANT_ENTITY);
+        AggregateIterable<Document> clusteredVariants = collection.aggregate(buildAggregation())
+                                                                  .allowDiskUse(true)
+                                                                  .useCursor(true);
+        cursor = clusteredVariants.iterator();
+    }
+
+    List<Bson> buildAggregation() {
+        Bson match = Aggregates.match(Filters.eq(REFERENCE_ASSEMBLY_FIELD, assemblyAccession));
+        Bson lookup = Aggregates.lookup(DBSNP_SUBMITTED_VARIANT_ENTITY, ACCESSION_FIELD,
+                                        CLUSTERED_VARIANT_ACCESSION_FIELD, SS_INFO_FIELD);
+        Bson sort = Aggregates.sort(orderBy(ascending(CONTIG_FIELD, START_FIELD)));
+        return Arrays.asList(match, lookup, sort);
+    }
+
+    @Override
+    public Variant read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+        return cursor.hasNext() ? getVariant(cursor.next()) : null;
+    }
+
+    Variant getVariant(Document clusteredVariant) {
+        String contig = clusteredVariant.getString(CONTIG_FIELD);
+        long start = clusteredVariant.getLong(START_FIELD);
+        long rs = clusteredVariant.getLong(ACCESSION_FIELD);
+        String reference = "";
+        String alternate = "";
+        long end = start;
+        List<VariantSourceEntry> sourceEntries = new ArrayList<>();
+        String type = clusteredVariant.getString(TYPE_FIELD);
+        String sequenceOntology = VariantTypeToSOAccessionMap.getSequenceOntologyAccession(VariantType.valueOf(type));
+
+        Collection<Document> submittedVariants = (Collection<Document>)clusteredVariant.get(SS_INFO_FIELD);
+        for (Document submitedVariant : submittedVariants) {
+            reference = submitedVariant.getString(REFERENCE_ALLELE_FIELD);
+            alternate = submitedVariant.getString(ALTERNATE_ALLELE_FIELD);
+            end = calculateEnd(reference, alternate, start);
+            String study = submitedVariant.getString(STUDY_FIELD);
+            sourceEntries.add(buildVariantSourceEntry(study, sequenceOntology));
+        }
+
+        Variant variant = new Variant(contig, start, end, reference, alternate);
```

Comment:
How does this work for multi-allelics (ex: rs1 that clusters ss1 -> G/T, ss2 -> G/A)? Won't we be writing only the ref and alt in the last SS?

Reply:
The code was changed in order to support multi-allelics. Variants will be grouped by `contig/start/reference/alternate`.